raiden/network/transport/matrix/client.py

```diff
@@ -282,12 +295,27 @@ def stop_listener_thread(self):
         self.should_listen = False
         if self.sync_thread:
             self.sync_thread.kill()
+            log.debug("Waiting on sync greenlet", current_user=self.user_id)
+            exited = gevent.wait([self.sync_thread], timeout=SHUTDOWN_TIMEOUT)
```

Comment:
please use `joinall` with `raise_error=True` and make the first argument a set

Reply:
Does it make sense to use joinall with raise_error here? The sync greenlet (and the handle one below as well) are `.link_exception()`ed to the transport.

Reply:
On second thought, there's no harm in using `joinall()`. I changed it to your suggestion.